test/shared/com/xilinx/rapidwright/support/CheckOpenFilesExtension.java

```diff
@@ -140,6 +141,9 @@ public static void assertExtensionInstalled(ExtensionContext extensionContext) {
     public static class CheckOpenFilesWorkingExtension implements AfterTestExecutionCallback {
         @Override
         public void afterTestExecution(ExtensionContext context) {
+            // if "gradle" not in command line ignore this test
+            boolean gradleStart = ManagementFactory.getRuntimeMXBean().getInputArguments().stream().anyMatch(n -> n.contains("gradle"));
+            Assumptions.assumeTrue(gradleStart);
```

Comment:
Two comments:
1. Would it be better to stick this assumption into the actual `TestCheckOpenFilesInstalled` invocation, as opposed to the callback which all (Gradle-executed) tests will call?
2. I wonder if there is a better way to check for "am I executed from Gradle"? (Let me do some research here too)

Reply:
Re: 2, I'm seeing three System properties that mention `gradle` and only exist when run from Gradle:
```
org.gradle.test.worker
org.gradle.native
org.gradle.internal.worker.tmpdir
```

Reply:
re 1: adding the assumption to TestCheckOpenFilesInstalled.test() doesn't work due to how Junit handles before/after anotations (https://github.com/junit-team/junit5/issues/2919#issuecomment-1132651012). This assumption was added to the afterTestExecution() method in the nested class CheckOpenFilesWorkingExtension. AFAIK Only TestCheckOpenFilesInstalled.test() is extended with CheckOpenFilesExtension.CheckOpenFilesWorkingExtension.class so only this test will run the gradle check; all other tests are auto-extended with CheckOpenFilesExtension.class which doesn't contain the overriding method with the assumption.

re 2: this is pretty janky, but I couldn't think of anything better (and simple). I think it might be possible to use some OS commands to get the parent process's PID and then convert that to a command line and check to see if the parent process is gradle but this is extremely not portable.

Reply:
Consider something like this?
```
public class TestCheckOpenFilesInstalled {
    @Test
    @EnabledIf("runningInGradle")
    @ExtendWith(CheckOpenFilesExtension.CheckOpenFilesWorkingExtension.class)
    public void test() throws IOException {
        //Actual test is in extension
    }

    private boolean runningInGradle() {
        return System.getProperty("org.gradle.test.worker") != null;
    }
}
```

Reply:
`@EnabledIf` certainly seems like a better option. Picking one system property is maybe a little less general than reading the whole command line but presumably there's no way a gradle test can be run without defining a worker thread.

Reply:
A more robust option would be to have our Gradle invocation define a custom property, and then detect that.